tests/format_utils/test_formats.cpp

```diff
@@ -0,0 +1,681 @@
+// Copyright 2023 The Khronos Group Inc.
+// Copyright 2023 Valve Corporation
+// Copyright 2023 LunarG, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+#define MAGIC_ENUM_RANGE_MIN 0
+#define MAGIC_ENUM_RANGE_MAX 512
+#include <magic_enum.hpp>
+#include <magic_enum_flags.hpp>
+#include <vulkan/utility/vk_format_utils.h>
+
+#include <string_view>
+
+// Given the string_view of a VkFormat, find the location of the letter that corresponds with a component
+// EG. find_component("R8G8B8", 'G') would return 2
+// returns std::string::npos if it is not found.
+size_t find_component(std::string_view str, char letter) {
+    size_t loc = str.find(letter, 10);
+    while (loc != std::string_view::npos) {
+        if (loc < str.length() - 1 && str[loc] == letter && (std::isdigit(str[loc - 1]) || str[loc - 1] == '_') &&
+            std::isdigit(str[loc + 1])) {
+            break;
+        }
+        loc = str.find(letter, loc + 1);
+    }
+
+    return loc;
+}
+
+// Given the string_view of a VkFormat, find the size of the letter that corresponds with a component
+// EG. find_component_size("R16G16B16", 3) would return 16
+// Returns 0 if the component can't be found
+size_t find_component_size(std::string_view str, char letter) {
+    size_t loc = find_component(str, letter);
+    if (loc == std::string_view::npos) return 0;
+    if (loc + 1 >= str.length()) return 0;
+    return std::stoi(std::string(str.substr(loc + 1)), nullptr);
+}
+
+size_t find_component_count(std::string_view str) {
+    size_t comp_count = 0;
+    if (find_component(str, 'R') != std::string_view::npos) comp_count++;
+    if (find_component(str, 'G') != std::string_view::npos) comp_count++;
+    if (find_component(str, 'B') != std::string_view::npos) comp_count++;
+    if (find_component(str, 'A') != std::string_view::npos) comp_count++;
+    if (find_component(str, 'D') != std::string_view::npos) comp_count++;
+    if (find_component(str, 'S') != std::string_view::npos) comp_count++;
+    return comp_count;
+}
+
+TEST(FormatUtils, FormatIsSFLOAT) {
+    for (auto [format, format_str] : magic_enum::enum_entries<VkFormat>()) {
+        // special case depth + stencil formats
+        if (format == VK_FORMAT_D32_SFLOAT_S8_UINT) {
+            EXPECT_FALSE(FormatIsSFLOAT(format));
+            continue;
+        }
```

Comment:
Is the rational because the stencil isn't SFLOAT? IE this function only returns true if each component of the format is SFLOAT?

Reply:
This was how the function worked before, I do not know how rational such a decision is. I guess it makes sense, but I think asking who ever added the function originally would be the best person.

Reply:
It's documented so it's fine. Perhaps having a FormatHasSFLOAT would be appropriate to have. But that a later feature request for another PR.

Reply:
I added this, there are some strange edge cases where one of these dual Depth/Stencil formats was returning true and we didn't want that behavior... I guess the `FormatHas` is just this but would return true for `D32S8` but not sure a use case... like if you are trying to match your Shader Interface, these dual formats are not allowed anyway

Reply:
>  but not sure a use case...

Right. I'm not sure there is a use case either. Which is why it's fine as is. If we need more functionality we can add it later.